packages/core/useElementBySelector/index.ts

```diff
@@ -0,0 +1,18 @@
+import type { MaybeRefOrGetter } from 'vue'
+import { tryOnMounted } from '@vueuse/shared'
+import { ref as deepRef, toValue, watch } from 'vue'
+
+export function useElementBySelector(selector: MaybeRefOrGetter<string>) {
+  const el = deepRef<Element | null>()
+  const select = (): Element | null => el.value = document.querySelector(toValue(selector))
+
+  tryOnMounted(select)
+
+  watch(
+    () => toValue(selector),
+    select,
+    { immediate: true },
```

Comment:
Please check out the Guidelines for watchers: https://vueuse.org/guidelines.html#watch-options

Reply:
Added